### ISSUE TITLE
Accessibility: Restore Focus to Last Focused Element after Closing Things like DynamicDialogs or Popups

### ISSUE THREAD

**Issue by Gykonik** [primefaces/primeng#13607]
### Describe the feature you would like to see added

I would like things like DynamicDialogs or Popups to automatically restore the focus to the last focused element after closing them. I think that this heavily impacts accessibility and should be the default behavior.

### Is your feature request related to a problem?

Yes, the problem is that when a DynamicDialog or Popup closes, the focus is not automatically restored to the last focused element which is bad for accessibility reasons. 

### Describe the solution you'd like

I would like to have a feature that automatically restores the focus to the last element that had focus before the DynamicDialog or Popup was opened (or similar components that change the focus). This should happen as soon as the dialog or popup is closed.

### Describe alternatives you have considered

I implemented a service manually doing this for all kinds of popups, but I feel like this should be the default behavior.

The service I'm currently using is the following:

```
@Injectable({
  providedIn: "root",
})
export class FocusService {
  private lastFocusedElement: HTMLElement | null = null;

  /** @function captureCurrentFocus - Capture the current focus. */
  public captureCurrentFocus(): void {
    this.lastFocusedElement = document.activeElement as HTMLElement;
  }

  /** @function restoreFocus - Restore the focus to the last focused element. */
  public restoreFocus(): void {
    if (this.lastFocusedElement) {
      this.lastFocusedElement.focus();
      this.lastFocusedElement = null;
    }
  }
}

```

**Comment by jasonverber** [comment]
We are doing the same -- seems like a good candidate for default behavior.

**Comment by mehmetcetin01140** [comment]
Hi,

So sorry for the delayed response! Improvements have been made to many components recently, both in terms of performance and enhancement. Therefore, this improvement may have been developed in another issue ticket without realizing it. You can check this in the documentation. If there is no improvement on this, can you open a new issue so we can include it in our roadmap?

Thanks a lot for your understanding!
Best Regards,
